src/invidious/routes/images.cr

```diff
@@ -181,29 +125,28 @@ module Invidious::Routes::Images
       end
     end
 
-    request_proc = ->(response : HTTP::Client::Response) {
-      env.response.status_code = response.status_code
-      response.headers.each do |key, value|
-        if !RESPONSE_HEADERS_BLACKLIST.includes?(key.downcase)
-          env.response.headers[key] = value
-        end
+    begin
+      get_ytimg_pool("i").client &.get(url, headers) do |resp|
+        return self.proxy_image(env, resp)
       end
+    rescue ex
+    end
+  end
 
-      env.response.headers["Access-Control-Allow-Origin"] = "*"
-
-      if response.status_code >= 300 && response.status_code != 404
-        return env.response.headers.delete("Transfer-Encoding")
+  private def self.proxy_image(env, response)
+    env.response.status_code = response.status_code
+    response.headers.each do |key, value|
+      if !RESPONSE_HEADERS_BLACKLIST.includes?(key.downcase)
+        env.response.headers[key] = value
       end
+    end
 
-      proxy_file(response, env)
-    }
+    env.response.headers["Access-Control-Allow-Origin"] = "*"
```

Comment:
I'm not sure actually. Do you want me to add a hook in the `proxy_image` to preserve this behavior?

Similar question with `#yts_image`, which has an extra here `!= 404` check here:

https://github.com/iv-org/invidious/blob/f23dd294ab49055204485d355b410689c604720b/src/invidious/routes/images.cr#L93-L95

Reply:
Yes, please preserve the previous behavior. I don't have the time atm to figure out if that code covers some edge case or if it can be safely removed!